src/auth/session/storage/mongodb.ts

```diff
@@ -75,6 +75,29 @@ export class MongoDBSessionStorage implements SessionStorage {
     return true;
   }
 
+  public async deleteSessions(ids: string[]): Promise<boolean> {
+    await this.ready;
+    for (const id of ids) {
+      await this.collection.deleteOne({id});
+    }
+    return true;
+  }
+
+  public async findSessionsByShop(
+    shop: string,
+  ): Promise<SessionInterface[] | {[key: string]: unknown}[] | undefined> {
+    await this.ready;
+
+    const rawResults = await this.collection.find().toArray();
```

Comment:
I'm not super familiar with the `find` method, but would it be possible to filter by the shop we're looking for at the query level, rather than filtering out the results? Seems like that would be more efficient.